connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -94,7 +94,7 @@ public DelegatingClassLoader(List<String> pluginPaths, ClassLoader parent) {
     }
 
     public DelegatingClassLoader(List<String> pluginPaths) {
-        this(pluginPaths, ClassLoader.getSystemClassLoader());
+        this(pluginPaths, DelegatingClassLoader.class.getClassLoader());
```

Comment:
It probably would have been good to add a comment with a short description of the assumptions that are made. Something like:
> Use the this classloader for flexibility. In most cases this will be System classloader, but it will also work in other managed environments that control classloading differently (OSGi, and others).

These kinds of choices are always difficult to track, especially when these comments are not in the commit description.